controllers/cluster_controller.go

```diff
@@ -640,31 +642,60 @@ func checkCoordinatorValidity(cluster *fdbtypes.FoundationDBCluster, status *fdb
 			continue
 		}
 
-		var address string
+		var ipAddress fdbtypes.ProcessAddress
 		for _, addr := range addresses {
 			if addr.Flags["tls"] == cluster.Spec.MainContainer.EnableTLS {
-				address = addr.String()
+				ipAddress = addr
 				break
 			}
 		}
 
-		_, isCoordinator := coordinatorStatus[address]
-		if isCoordinator && !process.Excluded && !pendingRemoval {
-			coordinatorStatus[address] = true
+		coordinatorAddress := ""
+		_, isCoordinatorWithIP := coordinatorStatus[ipAddress.String()]
+		if isCoordinatorWithIP {
+			coordinatorAddress = ipAddress.String()
 		}
 
-		if isCoordinator {
+		dnsName := process.Locality["dns_name"]
+		dnsAddress := fdbtypes.ProcessAddress{
+			StringAddress: dnsName,
+			Port:          ipAddress.Port,
+			Flags:         ipAddress.Flags,
+		}
+		_, isCoordinatorWithDNS := coordinatorStatus[dnsAddress.String()]
+
+		if !isCoordinatorWithDNS {
+			dnsAddress = ipAddress
+			dnsAddress.FromHostname = true
+			_, isCoordinatorWithDNS = coordinatorStatus[dnsAddress.String()]
```

Comment:
Is there a reason that we need the `127.0.0.1:4500(fromHostname)`? IMHO it doesn't make sense to have an IP address and define `fromHostname`.